storage/rocksdb/ha_rocksdb.cc

```diff
@@ -3687,15 +3686,66 @@ class Rdb_transaction {
 
   Rdb_io_perf *m_tbl_io_perf;
 
-  bool m_tx_read_only = false;
-
   int m_timeout_sec = 0; /* Cached value of @@rocksdb_lock_wait_timeout */
 
   /* Maximum number of locks the transaction can have */
   ulonglong m_max_row_locks;
 
   bool m_rollback_only = false;
 
+  enum class snapshot_type {
+    NONE,
+    // A snapshot created through Tranaction API for regular transactions and by
+    // DB::GetSnapshot() for WB ones
+    CURRENT,
+    CURRENT_DELAYED,
+    // This is used by transactions started with "START TRANSACTION WITH
+    // CONSISTENT [ROCKSDB] SNAPSHOT". The snapshot has to be created via
+    // DB::GetSnapshot(), not via Transaction API.
+    READ_ONLY_TRX,
```

Comment:
From rocksdb_explicit_snapshot, Looks like EXPLICIT snapshot also create via DB::GetSnapshot(). possible to merge these two?